File: Scouting-System-Server/src/main/kotlin/team8bit/org/Models/MatchStats.kt
package team8bit.org.Models

import kotlinx.serialization.Serializable //want to make our models serializable

@Serializable
data class MatchStats(
    val MatchNum: Int,
    val red1: Team,
    val red2: Team,
    val red3: Team,
    val blue1: Team,
    val blue2: Team,
    val blue3: Team
)

@Serializable
data class Team(
  var TeamNum: Int,
  var TeamName: String,
  var AvgSpeaker: Double,
  var AvgAmp: Double,
  var AvgClimb: Double,
  var AvgTrap: Double,
)

File: scouting-system-client/src/models/MatchStats.js
import { Team } from './Team';

export class MatchStats {
  constructor(matchNum, red1, red2, red3, blue1, blue2, blue3) {
    this.matchNum = matchNum;
    this.red1 = new Team(...Object.values(red1));
    this.red2 = new Team(...Object.values(red2));
    this.red3 = new Team(...Object.values(red3));
    this.blue1 = new Team(...Object.values(blue1));
    this.blue2 = new Team(...Object.values(blue2));
    this.blue3 = new Team(...Object.values(blue3));
  }
}